### PR TITLE
Addresses #243, default profile being used when it should not be

### LINKED ISSUE #243: Serious problem with default profile being used when it should not be. (Bugzilla #1010)
On 2012-12-19 11:27:48, @DavidGoldwasser wrote:

> Created an attachment (id=246)
> File from Eric
> 
> Eric found this working on the Fort Carson file for Matt. This only affects some of the schedules in the project. I found it on the activity schedule as well as setpoints, but didn't check everywhere else. As soon as I'm going to file this I am going to check the regression tests. I don't think it will have this issue, because it is hitting setopints, but I'm not sure it isn't affecting other schedules such as lighting or equipment that may not be as obvious. If it is affecting the regression then we may want to hold off 0.10.0. but if it is only affecting this file from Eric, then I would call it a blocker for 0.10.1.
> 
> Here is the text from the OSM file for the only rule in an activity schedule
> 
> OS:Schedule:Rule,
>   {fbe13c2a-e221-426d-872f-3bae56ef3a4f}, ! Handle
>   Medium Office_Activity Rule 1,          ! Name
>   {3a5bdc03-523b-4bf9-b0f8-9c4ec84616b0}, ! Schedule Ruleset Name
>   0,                                      ! Rule Order
>   {181ec819-81b1-49e4-9fe6-ce2ff0f8febc}, ! Day Schedule Name
>   Yes,                                    ! Apply Sunday
>   Yes,                                    ! Apply Monday
>   Yes,                                    ! Apply Tuesday
>   Yes,                                    ! Apply Wednesday
>   Yes,                                    ! Apply Thursday
>   Yes,                                    ! Apply Friday
>   Yes,                                    ! Apply Saturday
>   ,                                       ! Apply Holiday
>   DateRange,                              ! Date Specification Type
>   1,                                      ! Start Month
>   1,                                      ! Start Day
>   12,                                     ! End Month
>   31;                                     ! End Day
> 
> Here are two "Schedule:Week:Daily" objects in the exported IDF file. Notice that the first one is using the default profile for Sunday, Monday, Tuesday, and Wednesday.
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {0b23acb8-fedf-4ad9-bc59-88e565196ea4}, !- Name
>   Medium Office_Activity Default Schedule, !- Sunday Schedule:Day Name
>   Medium Office_Activity Default Schedule, !- Monday Schedule:Day Name
>   Medium Office_Activity Default Schedule, !- Tuesday Schedule:Day Name
>   Medium Office_Activity Default Schedule, !- Wednesday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Thursday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Friday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Saturday Schedule:Day Name
>   Medium Office_Activity Default Schedule, !- Holiday Schedule:Day Name
>   Medium Office_Activity Summer Design Day, !- SummerDesignDay Schedule:Day Name
>   Medium Office_Activity Winter Design Day, !- WinterDesignDay Schedule:Day Name
>   Medium Office_Activity Default Schedule, !- CustomDay1 Schedule:Day Name
>   Medium Office_Activity Default Schedule; !- CustomDay2 Schedule:Day Name
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {8aaa019f-148b-40cb-9915-d3a865505179}, !- Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Sunday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Monday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Tuesday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Wednesday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Thursday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Friday Schedule:Day Name
>   Medium Office_Activity Rule 1 Day Schedule, !- Saturday Schedule:Day Name
>   Medium Office_Activity Default Schedule, !- Holiday Schedule:Day Name
>   Medium Office_Activity Summer Design Day, !- SummerDesignDay Schedule:Day Name
>   Medium Office_Activity Winter Design Day, !- WinterDesignDay Schedule:Day Name
>   Medium Office_Activity Default Schedule, !- CustomDay1 Schedule:Day Name
>   Medium Office_Activity Default Schedule; !- CustomDay2 Schedule:Day Name
> 
> My first thought was that these are orphaned, but I am getting E+ warnings about these and we are missing setpoints for half of the year. Things look fine in the GUI, and open without any problems.

On 2012-12-19 11:28:42, @DavidGoldwasser wrote:

> Created an attachment (id=247)
> IDF file made from OSM

On 2012-12-19 11:29:04, @DavidGoldwasser wrote:

> Created an attachment (id=248)
> ABUPS

On 2012-12-19 11:29:25, @DavidGoldwasser wrote:

> Created an attachment (id=249)
> Error File

On 2012-12-19 11:49:56, @DavidGoldwasser wrote:

> I opened up the "baseline_sys01.osm" from the regression testing. It generally isn't testing the rules in the ruleset schedules. All schedules only use the "default' profile with one exception. "Baseline Model People Lights and Equipment Schedule" has a rule for Saturday, and then a rule for Sunday. Everything else should use the default. The IDF file reflects this (shown below). My next test is to make a new file from a template and test this.
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {8ae6feb6-c021-4356-b049-1a65ae2e48f3}, !- Name
>   Baseline Model People Lights and Equipment Schedule Schedule Sunday, !- Sunday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Monday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Tuesday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Wednesday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Thursday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Friday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Saturday, !- Saturday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Holiday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Summer Design Day, !- SummerDesignDay Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Winter Design Day, !- WinterDesignDay Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- CustomDay1 Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day; !- CustomDay2 Schedule:Day Name

On 2012-12-19 11:58:03, @DavidGoldwasser wrote:

> I need to do more testing, but this looks like a problem. From file made few days ago using Hotel Template. Seems like we might need a call to decide what to do about this for 0.10.0
> 
> OSM object
> 
> OS:Schedule:Rule,
>   {1227f732-d8a8-4f04-a5f5-f9250b7a259c}, ! Handle
>   LargeHotel_Work_Eff Rule 1,             ! Name
>   {ecdf8072-7fad-40d6-be54-c11aa4e4f8c5}, ! Schedule Ruleset Name
>   0,                                      ! Rule Order
>   {e489c4a6-939a-4058-acdf-0a1fc84a2d4e}, ! Day Schedule Name
>   Yes,                                    ! Apply Sunday
>   Yes,                                    ! Apply Monday
>   Yes,                                    ! Apply Tuesday
>   Yes,                                    ! Apply Wednesday
>   Yes,                                    ! Apply Thursday
>   Yes,                                    ! Apply Friday
>   Yes,                                    ! Apply Saturday
>   ,                                       ! Apply Holiday
>   DateRange,                              ! Date Specification Type
>   1,                                      ! Start Month
>   1,                                      ! Start Day
>   12,                                     ! End Month
>   31;                                     ! End Day
> 
> IDF objects made from this
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {e3a68135-953c-4d8d-b640-56014337f9d8}, !- Name
>   LargeHotel_Work_Eff Default Schedule,   !- Sunday Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule,   !- Monday Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule,   !- Tuesday Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule,   !- Wednesday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Thursday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Friday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Saturday Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule,   !- Holiday Schedule:Day Name
>   LargeHotel_Work_Eff Summer Design Day,  !- SummerDesignDay Schedule:Day Name
>   LargeHotel_Work_Eff Winter Design Day,  !- WinterDesignDay Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule,   !- CustomDay1 Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule;   !- CustomDay2 Schedule:Day Name
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {16a394b6-4a6e-4c1c-8334-966378e530c6}, !- Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Sunday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Monday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Tuesday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Wednesday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Thursday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Friday Schedule:Day Name
>   LargeHotel_Work_Eff Rule 1 Day Schedule, !- Saturday Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule,   !- Holiday Schedule:Day Name
>   LargeHotel_Work_Eff Summer Design Day,  !- SummerDesignDay Schedule:Day Name
>   LargeHotel_Work_Eff Winter Design Day,  !- WinterDesignDay Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule,   !- CustomDay1 Schedule:Day Name
>   LargeHotel_Work_Eff Default Schedule;   !- CustomDay2 Schedule:Day Name
> 
> It is weird, these come in pairs. Is it possible the top one isn't being used? but then why is it getting made. Hmm, the object below calls both of these.
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {fe40462e-b98f-4152-8bdd-3bfc32e408e0}, !- Name
>   LargeHotel_LaundryRoom_Gas Default Schedule, !- Sunday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule, !- Monday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule, !- Tuesday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule, !- Wednesday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Thursday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Friday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Saturday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule, !- Holiday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Summer Design Day, !- SummerDesignDay Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Winter Design Day, !- WinterDesignDay Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule, !- CustomDay1 Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule; !- CustomDay2 Schedule:Day Name
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {9e640e4d-c254-4e74-ad0d-8d4d7f2f322e}, !- Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Sunday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Monday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Tuesday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Wednesday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Thursday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Friday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Rule 1 Day Schedule, !- Saturday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule, !- Holiday Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Summer Design Day, !- SummerDesignDay Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Winter Design Day, !- WinterDesignDay Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule, !- CustomDay1 Schedule:Day Name
>   LargeHotel_LaundryRoom_Gas Default Schedule; !- CustomDay2 Schedule:Day Name
> 
> Schedule:Year,
>   LargeHotel_LaundryRoom_Gas,             !- Name
>   Fraction,                               !- Schedule Type Limits Name
>   Schedule:Week:Daily {fe40462e-b98f-4152-8bdd-3bfc32e408e0}, !- Schedule:Week Name 1
>   1,                                      !- Start Month 1
>   1,                                      !- Start Day 1
>   1,                                      !- End Month 1
>   3,                                      !- End Day 1
>   Schedule:Week:Daily {9e640e4d-c254-4e74-ad0d-8d4d7f2f322e}, !- Schedule:Week Name 2
>   1,                                      !- Start Month 2
>   4,                                      !- Start Day 2
>   12,                                     !- End Month 2
>   31;                                     !- End Day 2

On 2012-12-19 11:58:30, @DavidGoldwasser wrote:

> Created an attachment (id=250)
> This is the large hotel OSM from the last post

On 2012-12-19 12:07:52, @DavidGoldwasser wrote:

> Going back to Eric's file it has that same Schedule:Year object where the first three days of the year are using the bad "Schedule:Week"Daily" object
> 
> Schedule:Year,
>   Medium Office_Activity,                 !- Name
>   ActivityLevel,                          !- Schedule Type Limits Name
>   Schedule:Week:Daily {0b23acb8-fedf-4ad9-bc59-88e565196ea4}, !- Schedule:Week Name 1
>   1,                                      !- Start Month 1
>   1,                                      !- Start Day 1
>   1,                                      !- End Month 1
>   3,                                      !- End Day 1
>   Schedule:Week:Daily {8aaa019f-148b-40cb-9915-d3a865505179}, !- Schedule:Week Name 2
>   1,                                      !- Start Month 2
>   4,                                      !- Start Day 2
>   12,                                     !- End Month 2
>   31;                                     !- End Day 2

On 2012-12-19 12:39:27, @DavidGoldwasser wrote:

> Created an attachment (id=251)
> Screenshot of Laundry Gas in Results Viewer
> 
> Per Brent's request I requested the "Schedule Value" variable. Oddly it looks fine.

On 2012-12-19 13:18:43, @DavidGoldwasser wrote:

> I incorrectly stated earlier that the regression test didn't suffer from this. It does, and follows the same pattern. It doesn't have a profile named "default" but instead uses "Schedule Week Day" as the default. Notice how in the top of the two Schedule:Week:Daily objects "Schedule Week Day" is used in place of "Schedule Sunday". The ruby generated file has the same issue.
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {4277c26e-dd6b-4cac-bafc-0875d01baeeb}, !- Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Sunday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Monday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Tuesday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Wednesday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Thursday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Friday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Saturday, !- Saturday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Holiday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Summer Design Day, !- SummerDesignDay Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Winter Design Day, !- WinterDesignDay Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- CustomDay1 Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day; !- CustomDay2 Schedule:Day Name
> 
> Schedule:Week:Daily,
>   Schedule:Week:Daily {8ae6feb6-c021-4356-b049-1a65ae2e48f3}, !- Name
>   Baseline Model People Lights and Equipment Schedule Schedule Sunday, !- Sunday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Monday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Tuesday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Wednesday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Thursday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Friday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Saturday, !- Saturday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- Holiday Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Summer Design Day, !- SummerDesignDay Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Winter Design Day, !- WinterDesignDay Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day, !- CustomDay1 Schedule:Day Name
>   Baseline Model People Lights and Equipment Schedule Schedule Week Day; !- CustomDay2 Schedule:Day Name
> 
> Schedule:Year,
>   Baseline Model People Lights and Equipment Schedule, !- Name
>   Fractional,                             !- Schedule Type Limits Name
>   Schedule:Week:Daily {4277c26e-dd6b-4cac-bafc-0875d01baeeb}, !- Schedule:Week Name 1
>   1,                                      !- Start Month 1
>   1,                                      !- Start Day 1
>   1,                                      !- End Month 1
>   3,                                      !- End Day 1
>   Schedule:Week:Daily {8ae6feb6-c021-4356-b049-1a65ae2e48f3}, !- Schedule:Week Name 2
>   1,                                      !- Start Month 2
>   4,                                      !- Start Day 2
>   12,                                     !- End Month 2
>   31;                                     !- End Day 2

On 2012-12-19 13:43:14, @DavidGoldwasser wrote:

> Brent pointed out the weather file starts on Sunday, but our RunPeriod object in IDF shows Thursday. Which explains why the ResultsViewer for Laundry Gas wasn't affected (There were no Sunday's, Monday's Tuesday's, or Wednesday's between January 1-3).
> 
> So it seems like we are already ignoring the "Use Weather file for first day of year". Brent pointed out what day the year starts shouldn't affect how we right schedules. Users should be able to change the start day of year later in the IDF, and things should still work.
> 
> Epw file is Sunday, shouldn't matter. The Thursday comes from here:
> 
> RunPeriod,
>   Run Period 1,                           !- Name
>   1,                                      !- Begin Month
>   1,                                      !- Begin Day of Month
>   12,                                     !- End Month
>   31,                                     !- End Day of Month
>   Thursday,                               !- Day of Week for Start Day
>   Yes,                                    !- Use Weather File Holidays and Special Days
> 
> Another kind of scary thing is that globally we are using a default profile for "Holiday". Most of our template schedules have zeroed out default profile. With "Use Weather File Holidays and Special Days" I don't really know what is happening. Loads and people being "0" is fine, but doesn't make a nice setpoint.

On 2012-12-19 14:07:25, @kbenne wrote:

> This might be a secondary effect, but it appears the model is using an older sizing value that differs slightly from operation.  In the system sizing object for the air systems, Central Heating Design Supply Air Temperature, is 16.7 but the system is operating at 12.8.  The value of that system sizing field should be changed to 12.8 to match operation.  The latest generation of OpenStudio templates will provide correct values but previously created files will need to be updated.  In my experience this can cause air terminal heating coils to be undersized, because they are not prepared to provide the heating lift they need to in operation.

On 2012-12-21 10:24:23, @DavidGoldwasser wrote:

> **\* Bug 1011 has been marked as a duplicate of this bug. ***

On 2013-01-10 11:37:04, @DavidGoldwasser wrote:

> <batch comment: added this to Pivotal Epic, https://www.pivotaltracker.com/epic/show/458473>

Original attachments:
- 1218_FtCarson_a.osm (ID 246)
- out.idf (ID 247)
- eplustbl.htm (ID 248)
- eplusout.err (ID 249)
- 1210c.osm (ID 250)
- Jan1-Jan3_LaundryGass.png (ID 251)

Keywords: BadSimResults, ForwardTranslation

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Fixes #243
 - On forward translation of ScheduleRuleset to ScheduleYear, backfill the jan1 day schedule to all days of the week before it. This will ensure that any changes to the RunPeriod object in the idf will not cause the behavior observed in issue #243. This also comes with an additional ft unit test for ScheduleRuleset: ForwardTranslator_ScheduleWeek_Bug243.

Please read [OpenStudio Pull Requests](https://github.com/NREL/OpenStudio/wiki/OpenStudio-Pull-Requests) to better understand the OpenStudio Pull Request protocol.

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [x] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by shorowit** [comment]
@joseph-robertson Don't you have to do similar logic for the last week of the year?

**Comment by joseph-robertson** [comment]
Appears that this is already (inadvertently?) applied to the last week of the year. Here is a snippet from an idf with RunPeriod year set to 2015 (dec31 is a Thursday):
```
Schedule:Week:Daily,
  res cooling season Week Rule - Dec6-Dec31, !- Name
  res cooling season allday12,            !- Sunday Schedule:Day Name
  res cooling season allday12,            !- Monday Schedule:Day Name
  res cooling season allday12,            !- Tuesday Schedule:Day Name
  res cooling season allday12,            !- Wednesday Schedule:Day Name
  res cooling season allday12,            !- Thursday Schedule:Day Name
  res cooling season allday12,            !- Friday Schedule:Day Name
  res cooling season allday12,            !- Saturday Schedule:Day Name
  Schedule Day 1,                         !- Holiday Schedule:Day Name
  Schedule Day 1,                         !- SummerDesignDay Schedule:Day Name
  Schedule Day 1,                         !- WinterDesignDay Schedule:Day Name
  Schedule Day 1,                         !- CustomDay1 Schedule:Day Name
  Schedule Day 1;                         !- CustomDay2 Schedule:Day Name
```

File: src/energyplus/Test/ScheduleRuleset_GTest.cpp [hunk 1]
```diff
@@ -268,4 +271,39 @@ TEST_F(EnergyPlusFixture, ForwardTranslator_ScheduleWeek_Bug2322)
   //std::stringstream ss;
   //ss << workspace;
   //std::string s = ss.str();
+}
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_ScheduleWeek_Bug243)
+{
+  Model model;
+  ScheduleRuleset scheduleRuleset(model);
```
Comment (jmarrec) [inline src/energyplus/Test/ScheduleRuleset_GTest.cpp]: Nice test!

For clarity it'd be nice to rename objects

```
  ScheduleRuleset scheduleRuleset(model);
  scheduleRuleset.setName("Schedule Ruleset");
  ScheduleDay defaultDaySchedule = scheduleRuleset.defaultDaySchedule();
  defaultDaySchedule.setName("Default Day Schedule");
```

File: src/energyplus/Test/ScheduleRuleset_GTest.cpp [hunk 2]
```diff
@@ -268,4 +271,39 @@ TEST_F(EnergyPlusFixture, ForwardTranslator_ScheduleWeek_Bug2322)
...
+
+  ASSERT_EQ(1u, model.getModelObjects<ScheduleRuleset>().size());
+  ASSERT_EQ(1u, model.getModelObjects<ScheduleDay>().size());
+
+  ScheduleRule rule1(scheduleRuleset);
```
Comment (jmarrec) [inline src/energyplus/Test/ScheduleRuleset_GTest.cpp]:
```
  rule1.setName("All Days Rule");
  ScheduleDay ruleSchedule = rule1.daySchedule();
  ruleSchedule.setName("All Days Schedule");
```

File: src/energyplus/Test/ScheduleRuleset_GTest.cpp [hunk 3]
```diff
@@ -268,4 +271,39 @@ TEST_F(EnergyPlusFixture, ForwardTranslator_ScheduleWeek_Bug2322)
...
+  ASSERT_EQ(1u, model.getModelObjects<ScheduleRule>().size());
+  ASSERT_EQ(2u, model.getModelObjects<ScheduleDay>().size());
+
+  ForwardTranslator ft;
+  Workspace workspace = ft.translateModel(model);
+
+  ASSERT_EQ(1u, workspace.getObjectsByType(IddObjectType::Schedule_Year).size());
+  ASSERT_EQ(1u, workspace.getObjectsByType(IddObjectType::Schedule_Week_Daily).size());
+  EXPECT_EQ("Schedule Ruleset 1 Week Rule - Jan1-Dec31", workspace.getObjectsByType(IddObjectType::Schedule_Week_Daily)[0].nameString());
+  EXPECT_EQ("Schedule Day 2", workspace.getObjectsByType(IddObjectType::Schedule_Week_Daily)[0].getString(Schedule_Week_DailyFields::SundaySchedule_DayName).get());
```
Comment (jmarrec) [inline src/energyplus/Test/ScheduleRuleset_GTest.cpp]: Either `"All Days Schedule"` or `ruleSchedule.nameString()`

**Comment by jmarrec** [comment]
@joseph-robertson Made minor changes to your test.

**Comment by jmarrec** [comment]
Extended Gtests, and added a proposed solution (that we can revert if need be).

```ruby
m = Model.new

yd = m.getYearDescription
yd.setDayofWeekforStartDay("Wednesday")

scheduleRuleset = ScheduleRuleset.new(m)
scheduleRuleset.setName("Schedule Ruleset")
scheduleRuleset.defaultDaySchedule.setName("Default Day Schedule")

rule = ScheduleRule.new(scheduleRuleset)
rule.setName("All Year Weekdays Rule")
rule.daySchedule.setName("All Days but Monday and Wednesday Schedule")
rule.setApplyMonday(false)
rule.setApplyTuesday(true)
rule.setApplyWednesday(false)
rule.setApplyThursday(true)
rule.setApplyFriday(true)
rule.setApplySaturday(true)
rule.setApplySunday(true)
rule.setStartDate(OpenStudio::Date.new(OpenStudio::MonthOfYear.new(1), 3))
rule.setEndDate(OpenStudio::Date.new(OpenStudio::MonthOfYear.new(12), 29))

yd.makeDate(1, 1).dayOfWeek.valueName == "Wednesday"
yd.makeDate(12, 31).dayOfWeek.valueName == "Wednesday"



ft = OpenStudio::EnergyPlus::ForwardTranslator.new
w = ft.translateModel(m)

puts w.getObjectsByType("Schedule:Week:Daily".to_IddObjectType)
```

Overall, I don't think this was a big issue to begin with. And we'll always have some "trouble" to support changing the RunPeriod day of week / year after FT, since we use a Schedule:Week:Daily object... The placement of schedules for the first and last days depends on the year/start day

**Comment by joseph-robertson** [comment]
@jmarrec I think any days before Jan 1 should have what's on Jan 1. And any days after Dec 31 should have what's on Dec 31. Is that not what's happening with this PR?

**Comment by jmarrec** [comment]
I think I misunderstood the initial problem which lead me to another solution. Please revert back to your original solution if you think it's more appropriate, but consider keeping the extra GTests as more tests is a good thing (and adjusting as per what's actually expected)

**Comment by joseph-robertson** [comment]
This should be ready to merge now.

**Pull request merged**